### PR TITLE
Replace deprecated cgi module with multipart module

### LINKED ISSUE #732: cgi is deprecated and slated for removal in Python 3.13
Pytest on Python 3.11 pre-release will generate...
```
web/webapi.py:6: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi
```
https://docs.python.org/3/library/cgi.html

### PR DESCRIPTION
**Pull request by tfmorris**
May fix #732 

I took a cut at replacing the deprecated cgi module. The tests are green, but I don't really trust their coverage, so someone (not me) should give it a fair bit of testing.

I also removed a usage of the deprecated Thread.getName() function as a completely unrelated bonus commit.

### CODE REVIEW

**Comment by tfmorris** [comment]
Argh!! There's already a PR #768 open for this, but it's not linked to the issue #732. Why do people not follow the most basic of development practices?!?!?

File: web/webapi.py [hunk 1]
```diff
@@ -420,7 +421,9 @@ def dictify(fs):
                 # since wsgi.input is directly passed to multipart,
                 # it can not be called multiple times. Saving the result
                 # object in ctx to allow calling web.input multiple times.
-                a = ctx.get("_fieldstorage")  # TODO: Rename? is this visible anywhere else?
+                a = ctx.get(
+                    "_fieldstorage"
+                )  # TODO: Rename? is this visible anywhere else?
```
Comment (cclauss) [inline web/webapi.py:426]: Please do not wrap lines just to make room for a comment.
```suggestion
                # TODO: Rename?  Is this visible anywhere else?
                a = ctx.get("_fieldstorage")
```

File: web/webapi.py [hunk 2]
```diff
@@ -438,7 +441,9 @@ def dictify(fs):
     def process_values(values):
         if isinstance(values, list):
             return [process_values(x) for x in values]
-        elif hasattr(values, "filename") and values.filename is None:  # FIXME this probably needs to be improved
+        elif (
+            hasattr(values, "filename") and values.filename is None
+        ):  # FIXME this probably needs to be improved
```
Comment (cclauss) [inline web/webapi.py:446]: Same.

File: web/webapi.py (hunk 1 above)
Comment (tfmorris) [reply web/webapi.py:426]: Those changes were made by `ruff --fix`, not me.

**Pull request closed without merging**